pkg/nodebootstrap/userdata.go

```diff
@@ -77,6 +86,21 @@ func NewManagedBootstrapper(clusterConfig *api.ClusterConfig, ng *api.ManagedNod
 	return nil
 }
 
+func getClusterDNS(clusterConfig *api.ClusterConfig) (string, error) {
```

Comment:
is this path tested?

Reply:
Yes: https://github.com/weaveworks/eksctl/pull/3867/files#diff-7fffb935439e217b3a7c5100d1de60b26e2d8c71d525e80c6c00bf477b3a9bb0R135

Reply:
can we add some more coverage? it looks like that test is only doing happy path, and that it ends up in the userdata so it is technically testing 2 things.

Or maybe move the test of this to another location? like `pkg/nodebootstrap/userdata_test.go` or smth. The DNS thing will run for every call of `NewBootstrapper`, not in the `UserData` call, and not for just al2.
We didn't have tests for `NewBootstrapper` before since it wasn't really doing anything. Now it actually does something, we can have a test (or 3 to cover all cases introduced by this func) at a different "level".

Reply:
Makes sense, there does need to be a dedicated test for cluster DNS.